src/EnergyPlus/SurfaceGeometry.cc

```diff
@@ -9475,54 +9532,59 @@ namespace SurfaceGeometry {
             auto &vertices = surface.Vertex;
             auto &nSides = surface.Sides;
 
-            bool poppedVertex = true;
-            while (poppedVertex) {
-                poppedVertex = false;
-                Perimeter = 0.0;
+            while (true) {
+                PopCoincidentVertexReturn const popResult = checkPopCoincidentVertex(vertices);
+                Perimeter = popResult.perimeter;
+                if (popResult.poppedVertexPos < 0) {
+                    // No pop needed, we're done
+                    break;
+                }
 
-                for (auto it = vertices.begin(); it != vertices.end(); ++it) {
-                    auto itnext = std::next(it);
-                    if (itnext == std::end(vertices)) {
-                        itnext = std::begin(vertices);
-                    }
+                auto it = vertices.begin();
+                std::advance(it, popResult.poppedVertexPos);
+                int const poppedVertexIndex = popResult.poppedVertexPos + 1;
 
-                    // TODO: use isAlmostEqual3Pt for consistency? (which uses 0.0127 m / 1/2inch instead of 0.01 m)
```

Comment:
Kind of a side note but we really should use a single tolerance in  the code.

https://github.com/NREL/EnergyPlus/blob/50b8a5511ce559e5175524b943c26cc5b99d712d/src/EnergyPlus/SurfaceGeometry.cc#L13091

and other places

Reply:
That TODO was left by me last time I was in there. So yes, I agree we probably should.